src/Browser.php

```diff
@@ -720,7 +721,7 @@ private function withOptions(array $options)
      * @param string                         $url
      * @param array                          $headers
      * @param string|ReadableStreamInterface $body
-     * @return PromiseInterface<ResponseInterface,Exception>
+     * @return PromiseInterface<ResponseInterface,Exception, ConnectionInterface>
```

Comment:
I added it in the case, where an `upgrade` did happen, the promise will resolve with a `ConnectionInterface` rather than an `Exception` or a `ResponseInterface`. Still not sure if this is the best way to express that.